src/servers/buttongestureadapter.ts

```diff
@@ -0,0 +1,128 @@
+import {
+    ButtonEvent,
+    ButtonGestureEvent,
+    CONNECT,
+    DEVICE_CHANGE,
+    DEVICE_CONNECT,
+    EVENT,
+    ROLE_BOUND,
+    SELF_ANNOUNCE,
+    SRV_BUTTON,
+    SRV_BUTTON_GESTURE,
+} from "../jdom/constants"
+import SensorServer from "./sensorserver"
+import { assert, JDDevice, JDEvent, JDService } from "../jdom/jacdac-jdom"
+import RoleManager from "./rolemanager"
+import JDServiceServer from "../jdom/serviceserver"
+
+export class AdapterServer extends JDServiceServer {
+    private roleManager? : RoleManager = undefined
+    
+    // This is a nasty hack that allows the role manager to be set after the server is instantiated.
+    // The right solution would be to have this listen for role broadcasts for something
+    // but that's not a now feature.
+    // TODO de-jankify
+    public _hack_setRoleManager(roleManager: RoleManager) {
+        assert(this.roleManager === undefined, "resetting role manager")
+
+        this.roleManager = roleManager
+        this.onRoleManager(roleManager)
+    }
+
+    // to be overloaded - code to run once a role manager is available
+    protected onRoleManager(roleManager: RoleManager) { }
```

Comment:
Use on/subscribe instead

Reply:
Is there an event for getting the bus role manager?

Also, `DEVICE_CHANGE` only seems to provide `this.device`, but not `this.device.bus` which (I assume - from #218?) I need to listen for `ROLE_MANAGER_CHANGE` (?). I don't see any events emitted in ServiceProvider (`this.device`) when its `_bus` is set?